heareval/predictions/task_predictions.py

```diff
@@ -320,96 +342,117 @@ def dataloader_from_split_name(
 
 
 def task_predictions_train(
-    embedding_path: Path, scene_embedding_size: int, timestamp_embedding_size: int
+    embedding_path: Path,
+    embedding_size: int,
+    metadata: Dict[str, Any],
+    label_to_idx: Dict[str, int],
+    nlabels: int,
+    scores: List[ScoreFunction],
 ) -> torch.nn.Module:
-    metadata = json.load(embedding_path.joinpath("task_metadata.json").open())
-    label_vocal, nlabels = label_vocab_nlabels(embedding_path)
+    predictor = PredictionModel(
+        embedding_size, label_to_idx, nlabels, metadata["prediction_type"], scores
+    )
+
+    # First score is the target
+    target_score = f"val_{str(scores[0])}"
+
+    checkpoint_callback = ModelCheckpoint(monitor=target_score, mode="max")
+    early_stop_callback = EarlyStopping(
+        monitor=target_score, min_delta=0.00, patience=10, verbose=False, mode="max"
+    )
 
-    if metadata["embedding_type"] == "scene":
-        predictor = PredictionModel(
-            scene_embedding_size, nlabels, metadata["prediction_type"]
-        )
-    elif metadata["embedding_type"] == "event":
-        predictor = PredictionModel(
-            timestamp_embedding_size, nlabels, metadata["prediction_type"]
-        )
-    else:
-        raise ValueError(f"embedding_type {metadata['embedding_type']} unknown.")
     # train on CPU
     # TODO: FIXME
```

Comment:
Is this still broken?

Reply:
yes. we still aren't giving the user any option to specify what device they want for training. we're still missing model selection, etc.